### PR TITLE
Quang Minh home work lesson 3

### PR DESCRIPTION
**Pull request by nquangminhBK**
This is my homework for lesson 3,
I have a problem that I did not find the most effecient soluton yet, which is " Sort Array by Increasing Frequency", and I'm still working on it. 
Please review my orther problem. Thank you

### CODE REVIEW

File: Homework/Quang_Minh/Lesson3/SortArrayByInCreasingFreq.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        int index = 0;
+        for (int element : frequencyMap.keySet()) {
+            Pair<Integer, Integer> temp = new Pair<>(element, frequencyMap.get(element));
+            list[index] = temp;
+            index++;
+        }
+        for (int i = 0; i < list.length; i++) {
+            System.out.print(list[i] + " ");
+        }
+        for (int i = 0; i < list.length - 1; i++) {
```
Comment (quangh33) [inline Homework/Quang_Minh/Lesson3/SortArrayByInCreasingFreq.java:28]: can use built-in sort func of java with custom comparator.

File: Homework/Quang_Minh/Lesson3/KthLargestElement.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public static int quickSelect(int[] nums, int k, int start, int end) {
+        if (nums.length == 1) return nums[0];
+        int indexPivot = partition(nums, start, end);
+        if (indexPivot == k) return nums[indexPivot];
+        if (indexPivot > k) return quickSelect(nums, k, start, indexPivot - 1);
+        return quickSelect(nums, k, indexPivot + 1, end);
+    }
+
+    public static int partition(int[] array, int start, int end) {
+        int random = array[end];
```
Comment (quangh33) [inline Homework/Quang_Minh/Lesson3/KthLargestElement.java:18]: doesn't look like a random pivot

File: Homework/Quang_Minh/Lesson3/KClosetPointsToOrigin.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public static int[] quickSelect(int[][] nums, int k, int start, int end) {
+        if (nums.length == 1) return nums[0];
+        int indexPivot = partition(nums, start, end);
+        if (indexPivot == k) return nums[indexPivot];
+        if (indexPivot > k) return quickSelect(nums, k, start, indexPivot - 1);
+        return quickSelect(nums, k, indexPivot + 1, end);
+    }
+
+    public static int partition(int[][] array, int start, int end) {
+        int[] random = array[end];
```
Comment (quangh33) [inline Homework/Quang_Minh/Lesson3/KClosetPointsToOrigin.java:21]: ditto

File: Homework/Quang_Minh/Lesson3/EvaluateReversePolish.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+public class EvaluateReversePolish {
+    public int evalRPN(String[] tokens) {
+        Stack<Integer> stack = new Stack<>();
+        for (String token : tokens) {
+            int tempResult;
+            switch (token) {
+                case "+": {
+                    tempResult = stack.pop() + stack.pop();
+                    stack.push(tempResult);
```
Comment (quangh33) [inline Homework/Quang_Minh/Lesson3/EvaluateReversePolish.java:13]: duplicated code

**Comment by nquangminhBK** [comment]
About 2 problem that use quick select, because I use Lomuto’s Partition Scheme, so the pivot must be the element in the last position of each sub array, and my fault is named the variable as "random", I fixed the var name

**Pull request merged**